plugins/file-highlight/prism-file-highlight.js

```diff
@@ -94,12 +192,12 @@
 				return a;
 			});
 		}
-
 	};
 
-	document.addEventListener('DOMContentLoaded', function () {
-		// execute inside handler, for dropping Event as argument
-		self.Prism.fileHighlight();
-	});
-
-})();
+	if (document.readyState === 'loading') {
+		// Loading hasn't finished yet
+		document.addEventListener('DOMContentLoaded', function () { Prism.fileHighlight(); });
+	} else {
+		Prism.fileHighlight();
```

Comment:
This is simply best practices when listening for DOMContentLoaded as can be referenced here [Checking_whether_loading_is_already_complete](https://developer.mozilla.org/en-US/docs/Web/Events/DOMContentLoaded#Checking_whether_loading_is_already_complete)
developer.mozilla.org. In our case it is in fact unlikely this condition will apply but I included it in the plugin source of my code because it was separate from the core.

Reply:
I would also like to hear someone elses opinion though, as I am very green to Prism core, although it seems rather straightforward. I would hate to assume.

Reply:
After #1974 this won't be an issue anymore.